src/tzdata/compile.jl

```diff
@@ -118,41 +133,62 @@ end
 isflag(flag::Char) = flag in ('w', 'u', 's')
 
 """
-    tryparse_dayofmonth(str::AbstractString) -> Union{Function,Nothing}
+    tryparse_dayofmonth_function(str::AbstractString) -> Union{Function,Nothing}
 
-Parse the various day-of-month formats used within tzdata source files.
+Parse the various day-of-month formats used within tzdata source files. Returns a function
+which generates a `Date` observing the rule. The function returned (`f`) can be called by
+providing a year and month arguments or a `Date` (e.g. `f(year, month)` or `f(::Date)`).
 
 ```julia
-julia> tryparse_dayofmonth("lastSun")
-last_sunday (generic function with 1 method)
+julia> f = tryparse_dayofmonth_function("lastSun")
+last_sunday_of_month (generic function with 1 method)
 
-julia> tryparse_dayofmonth("Sun>=8")
-#15 (generic function with 1 method)
+julia> f(2019, 3)
+2019-03-31
 
-julia> TimeZones.TZData.tryparse_dayofmonth("15")
+julia> f = tryparse_dayofmonth_function("Sun>=8")
 #16 (generic function with 1 method)
+
+julia> f(2019, 3)
+2019-03-10
+
+julia> f = tryparse_dayofmonth_function("Fri<=1")
+#16 (generic function with 1 method)
+
+julia> f(2019, 4)
+2019-03-29
+
+julia> f = tryparse_dayofmonth_function("15")
+#18 (generic function with 1 method)
+
+julia> f(2019, 3)
+2019-03-15
 ```
 """
-function tryparse_dayofmonth(str::AbstractString)
+function tryparse_dayofmonth_function(str::AbstractString)
     if occursin(r"^last\w{3}$", str)
         # We pre-built these functions above
         # They follow the format: "lastSun", "lastMon", etc.
-        LAST_DAY_OF_WEEK[str]
-    elseif (m = match(r"^(?<dow>\w{3})(?<op>[<>]=)(?<dom>\d{1,2})$", str)) !== nothing
+        LAST_WEEKDAY_OF_MONTH[str]
```

Comment:
Were we just using the wrong function for what this regex actually is?

Reply:
No, what was changed here was:
- `LAST_DAY_OF_WEEK` was renamed to `LAST_WEEKDAY_OF_MONTH`
- Part of the regex `[<>]=` was changed to `<=|>=` to try and be clearer

The diff is doing a bad job here

Reply:
Yeah I guess my question is more so why did we rename `LAST_DAY_OF_WEEK` to `LAST_WEEKDAY_OF_MONTH` as the functionality they imply is different

Reply:
I think I understand the confusion. The original constant `LAST_DAY_OF_WEEK` was incorrectly named and should have been called: `IS_LAST_WEEKDAY_OF_MONTH`. Additionally, the function was updated to  return a `Date` instead of a `Bool` which is why the `IS` prefix was dropped